core/elf.ml

```diff
@@ -148,6 +167,72 @@ let find_symbol t name =
       None)
 ;;
 
+let find_selection t name : Selection.t option =
+  let find_line_selection name =
+    let split_name = String.split name ~on:':' in
+    let desired_filename = List.nth split_name 0 in
+    let desired_line =
+      Option.try_with (fun () -> List.nth_exn split_name 1 |> Int.of_string)
+    in
+    let desired_col =
+      Option.try_with (fun () -> List.nth_exn split_name 2 |> Int.of_string)
+    in
+    let cols = ref [] in
+    traverse_debug_line
+      ~f:(fun header state ->
+        let filename = Owee_debug_line.get_filename header state in
+        let line = state.line in
+        let col = state.col in
+        match desired_filename, desired_line, filename with
+        | Some desired_filename, Some desired_line, Some filename ->
+          if String.(desired_filename = filename) && desired_line = line
+          then cols := (col, state.address) :: !cols
+        | None, _, _ | _, None, _ | _, _, None -> ())
+      t;
+    let cols =
+      List.sort !cols ~compare:(fun (col1, _) (col2, _) -> Int.compare col1 col2)
+    in
+    match cols with
+    | [] -> None
+    | (col, address) :: _ ->
+      (match desired_col with
+      | None ->
+        if List.length cols > 1
+        then
+          Core.eprint_s
+            [%message
+              "Multiple snapshot symbols on same line. Selecting one."
+                ~columns:(List.map ~f:Tuple2.get1 cols : int list)
+                ~selected:(col : int)];
+        Some (Selection.Address { address; name })
+      | Some desired_col ->
+        (match
+           List.find_map cols ~f:(fun (col, _) ->
+               if col = desired_col then Some address else None)
+         with
+        | None -> None
+        | Some address -> Some (Selection.Address { address; name })))
+  in
+  let find_addr_selection name =
+    Option.bind
+      (Option.try_with (fun () -> Int.Hex.of_string name))
+      ~f:(fun address -> Some (Selection.Address { address; name }))
+  in
+  let find_symbol_selection name =
+    Option.map (find_symbol t name) ~f:(fun symbol -> Selection.Symbol symbol)
+  in
+  if String.is_prefix name ~prefix:"symbol:"
```

Comment:
Let's extract this if/then branch into a short helper and use it here, that takes `~prefix` and `~f`, so that we can elide the hardcoding of the prefix lengths.